sql/playground.sql

```diff
@@ -0,0 +1,89 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- This function allows users to create playground hypertables with production-like data for feature testing.
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.create_playground(
+    src_hypertable REGCLASS,
+    compressed BOOL = false
+) RETURNS TEXT AS
+$BODY$
+DECLARE
+    _table_name NAME;
+    _schema_name NAME;
+    _src_relation NAME;
+    _playground_table NAME;
+    _chunk_name NAME;
+    _chunk_check BOOL;
+    _next_id INTEGER;
+    _dimension TEXT;
+    _interval TEXT;
+BEGIN
+    -- get schema and table name
+    SELECT n.nspname, c.relname INTO _schema_name, _table_name
+    FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.oid = c.relnamespace)
+    INNER JOIN timescaledb_information.hypertables i ON (i.hypertable_name = c.relname )
+    WHERE c.oid = src_hypertable;
+
+    IF _table_name IS NULL THEN
+        RAISE EXCEPTION '% is not a hypertable', src_hypertable;
+    END IF;
+
+    SELECT EXISTS(SELECT 1 FROM timescaledb_information.chunks WHERE hypertable_name = _table_name AND hypertable_schema = _schema_name) INTO _chunk_check;
+
+    IF NOT _chunk_check THEN
+        RAISE EXCEPTION '% has no chunks for playground testing', src_hypertable;
+    END IF;
+
+    CREATE SCHEMA IF NOT EXISTS tsdb_playground;
```

Comment:
Could you add this as a precondition -i.e. check for a schema named tsdb_playground and throw an error if it does not exist?

Reply:
Alternately, 1) we can create a pre-existing schema for this type of experimentation? 2) Or take a schema as an argument to the procedure.